assets/src/stories-editor/components/higher-order/with-right-click-handler.js

```diff
@@ -0,0 +1,105 @@
+/**
+ * WordPress dependencies
+ */
+import { withDispatch, withSelect } from '@wordpress/data';
+import { compose, createHigherOrderComponent } from '@wordpress/compose';
+
+/**
+ * Internal dependencies
+ */
+import { ALLOWED_CHILD_BLOCKS } from '../../constants';
+import { render } from '@wordpress/element';
+import { RightClickMenu } from '../';
+
+const applyWithSelect = withSelect( ( select ) => {
+	const {	isReordering } = select( 'amp/story' );
+	return {
+		isReordering: isReordering(),
+	};
+} );
+
+const applyWithDispatch = withDispatch( ( dispatch, ownProps, { select } ) => {
+	const {
+		getSelectedBlockClientIds,
+		hasMultiSelection,
+	} = select( 'core/block-editor' );
+
+	const onContextMenu = ( event ) => {
+		const selectedBlockClientIds = getSelectedBlockClientIds();
+
+		if ( selectedBlockClientIds.length === 0 ) {
+			return;
+		}
+
+		// Let's ignore multi-selection for now.
+		if ( hasMultiSelection() ) {
+			return;
```

Comment:
Can we also return early when text within a block has selected and not just that block as a whole?

Reply:
Added within c247c19.